### PR TITLE
Posts mc

### PR DESCRIPTION
**Pull request by marcho56**
post data and routes created

### CODE REVIEW

File: data/posts.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            throw 'title, and postContent parameters must be strings';
+        }
+        if (!(Array.isArray(tags) && tags.every(stringCheck)))
+            throw 'tags parameter must be an array of strings';
+        let user = users.findUser(id);
+        const postCollection = await posts();
+        let newPost = {
+            _postId: ObjectID(),
+            userId: userId,
+            userName: user.userName,
```
Comment (vodewey12) [inline data/posts.js]: we can query this with the userID from the app rather than doing it in the db

File: data/posts.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        let userId = idCheck(id);
+        if (!(stringCheck(title) && stringCheck(postContent))) {
+            throw 'title, and postContent parameters must be strings';
+        }
+        if (!(Array.isArray(tags) && tags.every(stringCheck)))
+            throw 'tags parameter must be an array of strings';
+        let user = users.findUser(id);
+        const postCollection = await posts();
+        let newPost = {
+            _postId: ObjectID(),
```
Comment (vodewey12) [inline data/posts.js]: `             _postId: ObjectID(), `
don't need to do this, since mongo automatically creates _id on its own. We can just that to query instead of this

File: data/posts.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            post._id = post._id.toString();
+        });
+        return postList;
+    },
+    async getPostById(id) {
+        let postId = idCheck(id);
+        const postCollection = await posts();
+        const post = await postCollection.findOne({ _postId: postId });
+        if (!post) throw 'post not found';
+        post._postId = post._postId.toString();
```
Comment (vodewey12) [inline data/posts.js]: `         post._postId = post._postId.toString(); `
This line causes an error when creating a posts. This is because you're already sending a tostring() id into the method when creating posts, and I guess you can tostring it twice. This also causes issues with updating a posts

tldr: remove this line

File: data/posts.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            if (typeof updatedPost.resolvedStatus !== 'boolean')
+                throw 'resolvedStatus attribute must be a boolean';
+            postUpdateInfo.resolvedStatus = updatedPost.resolvedStatus;
+        }
+        const postCollection = await posts();
+        const updateInfo = await postCollection.updateOne(
+            { _id: postId, postTime: JSON.stringify(new Date()) },
+            { $set: postUpdateInfo }
+        );
+        if (!updateInfo.matchedCount && !updateInfo.modifiedCount)
```
Comment (vodewey12) [inline data/posts.js]: !update.matchedCount is causing an error. 
`if (!updateInfo && !updateInfo.modifiedCount) throw "Update failed";`
This makes it work though.

File: data/posts.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            throw 'Update failed';
+
+        return await this.getPostById(postId.toString());
+    },
+    async deletePost(id) {
+        let postId = idCheck(id);
+        const postCollection = await posts();
+        const deletionInfo = await postCollection.removeOne({
+            _postId: postId
```
Comment (vodewey12) [inline data/posts.js]: `_id: postId`

Do this instead
